# Effective Arguments Predictor

Author: Xiaohua Su

Date: June, 2022

## Overview

## Problem

GSU and The Learning Agency Lab wants to create a predictive model that predicts whether an essay and its part is effective in order to better the quality of these automated tools. This is in hopes of providing better feedback and help students write more compelling arguments.

## Data

This data was take from the [Feedback Prize - Predicting Effective Arguments](https://www.kaggle.com/competitions/feedback-prize-effectiveness/overview) competition. In order to access the data, you must register for the competition and accept the rules and condition.

### Imports

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('./feedback-prize-effectiveness/train.csv')

In [3]:
df

discourse_id      essay_id  \
0      0013cc385424  007ACE74B050   
1      9704a709b505  007ACE74B050   
2      c22adee811b6  007ACE74B050   
3      a10d361e54e4  007ACE74B050   
4      db3e453ec4e2  007ACE74B050   
...             ...           ...   
36760  9f63b687e76a  FFA381E58FC6   
36761  9d5bd7d86212  FFA381E58FC6   
36762  f1b78becd573  FFA381E58FC6   
36763  cc184624ca8e  FFA381E58FC6   
36764  c8a973681feb  FFA381E58FC6   

                                          discourse_text  \
0      Hi, i'm Isaac, i'm going to be writing about h...   
1      On my perspective, I think that the face is a ...   
2      I think that the face is a natural landform be...   
3      If life was on Mars, we would know by now. The...   
4      People thought that the face was formed by ali...   
...                                                  ...   
36760  For many people they don't like only asking on...   
36761  also people have different views and opinions ...   
36762  Advice is something that can impact a persons ...   
36763  someone can use everything that many people sa...   
36764  In conclusion asking for an opinion can be ben...   

             discourse_type discourse_effectiveness  
0                      Lead                Adequate  
1                  Position                Adequate  
2                     Claim                Adequate  
3                  Evidence                Adequate  
4              Counterclaim                Adequate  
...                     ...                     ...  
36760                 Claim                Adequate  
36761                 Claim                Adequate  
36762              Position                Adequate  
36763              Evidence             Ineffective  
36764  Concluding Statement             Ineffective  

[36765 rows x 5 columns]

discourse_id - ID code for discourse element

essay_id - ID code for essay response. This ID code corresponds to the name of the full-text file in the train/ folder.

discourse_text - Text of discourse element.

discourse_type - Class label of discourse element.


discourse_type_num - Enumerated class label of discourse element .


discourse_effectiveness - Quality rating of discourse element, the target.

In [4]:
df.discourse_text[1]

"On my perspective, I think that the face is a natural landform because I dont think that there is any life on Mars. In these next few paragraphs, I'll be talking about how I think that is is a natural landform "

In [5]:
df[df['essay_id'] == '007ACE74B050']

discourse_id      essay_id  \
0  0013cc385424  007ACE74B050   
1  9704a709b505  007ACE74B050   
2  c22adee811b6  007ACE74B050   
3  a10d361e54e4  007ACE74B050   
4  db3e453ec4e2  007ACE74B050   
5  36a565e45db7  007ACE74B050   
6  fb65fe816ba3  007ACE74B050   
7  4e472e2584fa  007ACE74B050   
8  28a94d3ee425  007ACE74B050   

                                      discourse_text        discourse_type  \
0  Hi, i'm Isaac, i'm going to be writing about h...                  Lead   
1  On my perspective, I think that the face is a ...              Position   
2  I think that the face is a natural landform be...                 Claim   
3  If life was on Mars, we would know by now. The...              Evidence   
4  People thought that the face was formed by ali...          Counterclaim   
5  though some say that life on Mars does exist, ...              Rebuttal   
6  It says in paragraph 7, on April 5, 1998, Mars...              Evidence   
7  Everyone who thought it was made by alieans ev...          Counterclaim   
8  Though people were not satified about how the ...  Concluding Statement   

  discourse_effectiveness  
0                Adequate  
1                Adequate  
2                Adequate  
3                Adequate  
4                Adequate  
5             Ineffective  
6                Adequate  
7                Adequate  
8                Adequate

In [8]:
len(df.essay_id.unique())

4191

In [9]:
len(df.discourse_type.unique())

7

In [10]:
df.discourse_type.value_counts()

Evidence                12105
Claim                   11977
Position                 4024
Concluding Statement     3351
Lead                     2291
Counterclaim             1773
Rebuttal                 1244
Name: discourse_type, dtype: int64

In [11]:
df.discourse_effectiveness.value_counts()

Adequate       20977
Effective       9326
Ineffective     6462
Name: discourse_effectiveness, dtype: int64

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36765 entries, 0 to 36764
Data columns (total 5 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   discourse_id             36765 non-null  object
 1   essay_id                 36765 non-null  object
 2   discourse_text           36765 non-null  object
 3   discourse_type           36765 non-null  object
 4   discourse_effectiveness  36765 non-null  object
dtypes: object(5)
memory usage: 1.4+ MB


There are 4191 different essays, 7 different types of discourse in the training set, and the target having 3 different class. There is also no nulls within this dataset. 

## Data Preprocessing/Pipelines